src/Compilers/Core/Portable/CommandLine/AnalyzerConfigSet.cs

```diff
@@ -193,7 +193,7 @@ public AnalyzerConfigOptionsResult GetOptionsForSourcePath(string sourcePath)
                     treeOptionsBuilder.Count > 0 ? treeOptionsBuilder.ToImmutable() : SyntaxTree.EmptyDiagnosticOptions,
                     analyzerOptionsBuilder.Count > 0 ? analyzerOptionsBuilder.ToImmutable() : AnalyzerConfigOptions.EmptyDictionary,
                     diagnosticBuilder.ToImmutableAndFree());
-                _optionsCache.TryAdd(sectionKey, result);
+                _optionsCache.TryAdd(new List<Section>(sectionKey), result);
```

Comment:
>new List<Section>(sectionKey) [](start = 37, length = 29)

So, the problem was that were clearing and freeing the ```sectionKey``` below? If we wouldn't do that for this code path things would work?

Reply:
Would it make sense to ```ForgetTrackedObject``` on this code path instead?

---
In reply to: [343936912](https://github.com/dotnet/roslyn/pull/39729#discussion_r343936912) [](ancestors = 343936912)

Reply:
That would also work, but I figured that it doesn't matter either way -- if we have a stable pooled object count then we would then immediately allocate a new object for the pool to satisfy parallel requests.

So we either allocate a new object here, or in the pooling manager. This seemed less error-prone.

Reply:
> So we either allocate a new object here, or in the pooling manager. This seemed less error-prone.

It feels like delaying an allocation until it actually necessary is always good.

---
In reply to: [344299232](https://github.com/dotnet/roslyn/pull/39729#discussion_r344299232) [](ancestors = 344299232)

Reply:
Sounds fine to me